pkg/repository/local_git.go

```diff
@@ -36,13 +38,60 @@ var (
 
 type CommandExecutor interface {
 	RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error)
+	RunPipeCommands(ctx context.Context, dir string, cmd1 string, cmd1Args []string, cmd2 string, cmd2Args []string) ([]byte, error)
 	LookPath(file string) (string, error)
 }
 
 type LocalExec struct{}
 
-func (*LocalExec) RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error) {
-	c := exec.CommandContext(ctx, name, arg...)
+func (le *LocalExec) RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error) {
+	c := le.createCommand(ctx, dir, name, arg...)
+
+	return c.Output()
+}
+
+func (*LocalExec) LookPath(file string) (string, error) {
+	return exec.LookPath(file)
+}
+
+// RunPipeCommands runs two commands in a pipeline. cmd1 | cmd2. Example:
+// /usr/bin/git --no-optional-locks ls-files -s . | /usr/bin/git --no-optional-locks hash-object --stdin
+func (le *LocalExec) RunPipeCommands(ctx context.Context, dir string, cmd1 string, cmd1Args []string, cmd2 string, cmd2Args []string) ([]byte, error) {
+	c1 := le.createCommand(ctx, dir, cmd1, cmd1Args...)
+	c2 := le.createCommand(ctx, dir, cmd2, cmd2Args...)
+
+	// Create a pipe to connect the stdout of c1 to the stdin of c2
+	pipeReader, pipeWriter := io.Pipe()
+	c1.Stdout = pipeWriter
+	c2.Stdin = pipeReader
```

Comment:
It might be useful to set  `c1.Stderr = os.Stderr` (by default it goes to `/dev/null` I think)

Reply:
I have changed it to get the stderr to a buffer and log it if there is any error on any of the commands. @maroshii WDYT?

Reply:
👌